cli/src/vote.rs

```diff
@@ -907,6 +968,41 @@ pub fn process_withdraw_from_vote_account(
     log_instruction_custom_error::<VoteError>(result, config)
 }
 
+pub fn process_close_vote_account(
+    rpc_client: &RpcClient,
+    config: &CliConfig,
+    vote_account_pubkey: &Pubkey,
+    withdraw_authority: SignerIndex,
+    destination_account_pubkey: &Pubkey,
+    memo: Option<&String>,
+) -> ProcessResult {
+    let latest_blockhash = rpc_client.get_latest_blockhash()?;
+    let withdraw_authority = config.signers[withdraw_authority];
+
```

Comment:
I think `solana close-vote-account` should fail on vote accounts with active stake, since there are very few times when that would be desirable and it's most likely that somebody would accidentally close their staked vote account instead.

That check would look approximately like this (I didn't compile this snippet though, it might blow up!):
```rust
let vote_account_status = rpc_client.get_vote_accounts_with_config(
    RpcGetVoteAccountsConfig {
        vote_pubkey: Some(vote_account_pubkey),
        ..RpcGetVoteAccountsConfig::default()
    }
)?;
if let Some(vote_account) = 
        vote_account_status.current.into_iter().chain(vote_account_status.delinquent.into_iter()).next() {
    if vote_account.activated_stake != 0 {
        return Err(format!("Cannot close a vote account with active stake: {}", vote_account_pubkey).into());   
    }
} else {
    return Err(format!("Vote account does not exist: {}", vote_account_pubkey).into());
}
```

Reply:
sounds good - and thanks for catching this for me.

will look into it!